gate-core/src/main/groovy/com/netflix/spinnaker/gate/services/internal/ClouddriverService.groovy

```diff
@@ -293,6 +293,14 @@ interface ClouddriverService {
   @GET('/artifacts/credentials')
   List<Map> getArtifactCredentials()
 
+  @GET('/roles/{cloudProvider}')
+  List<Map> getRoles(@Path("cloudProvider") String cloudProvider)
```

Comment:
Is `/roles/{cloudProvider}` the right way to do this?

What roles are these? IAM roles for ECS?

Reply:
Yes, they are IAM roles that container tasks can inherit (as in, the trust relationship allows that).  Other cloud providers like EC2 and Google cloud do have roles, yet do not currently use them in any way.  They could if they wanted.

Code in clouddriver already caches all IAM roles and their trust relationships, so the EC2 cloud provider could easily have IAM role dropdowns